services/sso/src/routes/authenticationRouter.test.js

```diff
@@ -0,0 +1,49 @@
+jest.mock("passport");
+const passport = require("passport");
+const authenticationRouter = require("./authenticationRouter");
+const { mockReq, mockRes } = require("../mocks");
+
+describe("authenticationRouter", () => {
+  const mockStrategyId = "mock-strategy-id";
+  let req;
+  let res;
+
+  beforeEach(() => {
+    req = mockReq;
+    res = mockRes();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("should call passport with a mock strategy", () => {
+    authenticationRouter(req, res, mockStrategyId);
+    expect(passport.authenticate).toHaveBeenCalledWith(mockStrategyId, {
+      failureRedirect: "/",
+    });
+  });
+
+  it("should define the expected route with an existant user", () => {
+    req.user = { mockUser: "mock-user" };
+
+    authenticationRouter(req, res, mockStrategyId);
+    expect(res.redirect).toHaveBeenCalledWith("/logged-in");
+  });
+
+  it("should define the expected route with an non-existant user", () => {
```

Comment:
I believe non-existant should actually be spelt non-existent :)

Reply:
Edited, thank you.